### DISCUSSION TITLE
`.split()` output seems not recognised by `.transpose()`

### DISCUSSION THREAD

**Discussion by ktpolanski** [nextflow-io/nextflow#4074] [Q&A]
### Expected behavior and actual behavior

Performing a `.map()` which includes a `.split()` on one of the `it` components seems to create a form of data that does not get picked up and expanded by a subsequent `.transpose()`.

### Steps to reproduce the problem

```nextflow
ch1 = Channel.of(["A", ["B", "C"]])
ch1.transpose().view()

ch2 = Channel.of(["D", "E_F"])
ch2.map({it -> [it[0], it[1].split("_")]}).transpose().view()
```

### Program output 

```bash
N E X T F L O W  ~  version 23.04.1
Launching `transpose-test.nf` [scruffy_hodgkin] DSL2 - revision: 958c877451
[D, [E, F]]
[A, B]
[A, C]
```

### Environment 

* Nextflow version: 23.04.1 
* Java version: openjdk 17.0.6
* Operating system: Linux
* Bash version: `GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)`

**Comment by ktpolanski** [comment]
Of note, adding `.toList()` as part of the `.map()` seems to resolve this situation:

```nextflow
ch2 = Channel.of(["D", "E_F"])
ch2.map({it -> [it[0], it[1].split("_").toList()]}).transpose().view()
```

yields

```bash
N E X T F L O W  ~  version 23.04.1
Launching `transpose-test.nf` [insane_mirzakhani] DSL2 - revision: 28f818061d
[D, E]
[D, F]
```

**Comment by mribeirodantas** [comment]
This is not a bug, as you probably noticed with your second comment. The `split` function returns a string that may look like a list with two strings, but it's a full string. You can confirm this with the snippet below:

```Groovy
ch1 = Channel.of(["A", ["B", "C"]])
ch1.map { println it[1].getClass() }

ch2 = Channel.of(["D", "E_F"])
ch2.map { it -> [it[0], it[1].split("_")] }.map { println it[1].getClass() }
```

The output of the snippet above is:
```console
N E X T F L O W  ~  version 23.06.0-edge
Launching `foo.nf` [agitated_hilbert] DSL2 - revision: d5e6b1ed25
class java.util.ArrayList
class [Ljava.lang.String;
```

If you apply a `toList` to this string, then you have a list as the first case and things work accordingly :-)

**Comment by bentsherman** [comment]
Hi @ktpolanski , the `split()` method returns a string "array" (`String[]`) which is why you have to convert it to a list with `toList()`. Alternatively, you can use the `tokenize()` method which returns a regular list.

**Reply by ktpolanski** [reply]
`tokenize()` and `split()` technically [behave a bit differently](https://www.tothenew.com/blog/groovy-tokenize-vs-split/), potential users should be aware of this

```nextflow
ch2 = Channel.of("hello world")
ch2.map({it -> it.tokenize("lo")}).view()
ch2.map({it -> it.split("lo")}).view()
```

```
N E X T F L O W  ~  version 23.04.1
Launching `test.nf` [condescending_visvesvaraya] DSL2 - revision: d1db60d57a
[hel,  world]
[he,  w, r, d]
```
